lib/fastlane/plugin/appcenter/actions/appcenter_upload_action.rb

```diff
@@ -35,8 +42,9 @@ def self.run_dsym_upload(params)
 
         dsym_path = nil
         if dsym
-          # we can use dsym parameter only if build file is ipa
-          dsym_path = dsym if !file || File.extname(file) == '.ipa'
+          # we can use dsym parameter for all apple builds
+          self.optional_error("dsym parameter can only be used with Apple builds (ios, mac)") unless !file || self.is_apple_build(file)
+          dsym_path = dsym
         else
           # if dsym is not set, but build is ipa - check default path
           if file && File.exist?(file) && File.extname(file) == '.ipa'
```

Comment:
can this case also be covered , where dsym is not provided in the path but we checks the package folder and considers if dsym.zip exists

Reply:
I don't understand this comment, since automatic detection is covered by the `else` case. If you still have concerns about it, let's raise a PR later :)